zebra-consensus/src/error.rs

```diff
@@ -78,13 +78,17 @@ pub enum TransactionError {
     #[error("bindingSig MUST represent a valid signature under the transaction binding validating key bvk of SigHash")]
     RedJubjub(zebra_chain::primitives::redjubjub::Error),
 
+    #[error("bindingSig MUST represent a valid signature under the transaction binding validating key bvk of SigHash")]
+    RedPallas(zebra_chain::primitives::redpallas::Error),
+
     // temporary error type until #1186 is fixed
     #[error("Downcast from BoxError to redjubjub::Error failed")]
     InternalDowncastError(String),
 }
 
 impl From<BoxError> for TransactionError {
     fn from(err: BoxError) -> Self {
+        // TODO: handle redpallas Error?
         match err.downcast::<zebra_chain::primitives::redjubjub::Error>() {
```

Comment:
Yeah I think we should handle `redjubjub`, `redpallas`, and `tokio::Elapsed` (timeouts) here.

A good pattern is probably:
```rust
if let Ok(e) = err.downcast::<_>() {
    return TransactionError::_(*e);
}
```

Reply:
Let's also handle `ScriptInvalid` here:
https://github.com/ZcashFoundation/zebra/pull/2302/files#diff-dc6731c91623a1343ccf8cb5a86fc6aeadf942b3cc84435b2daa893c737fac59R449

Reply:
Idea: maybe this can be chained, something like:

```
err.downcast::<TransactionError>()
    .or_else(|err| err.downcast::<A>().map(TransactionError::ForA))
    .or_else(|err| err.downcast::<B>().map(TransactionError::ForB))
    .or_else(|err| err.downcast::<C>().map(TransactionError::ForC))
    .or_else(|_| TransactionError::Default)
```

Reply:
I've been trying to get this to work nicely but the `BoxError` and chaining the `Result`s and/or the `if/else` with `BoxError` not impl'ing Copy or Clone is giving me problems so I'm punting this for now